### ISSUE TITLE
Double softmax in cross entropy with logits

### ISSUE THREAD

**Issue by davidenunes** [nfmcclure/tensorflow_cookbook#86]
Just wanted to point out that you have some examples where you use 
`logits_out = tf.nn.softmax(tf.matmul(last, weight) + bias)`

and then define the loss function as:
`losses = tf.nn.sparse_softmax_cross_entropy_with_logits(logits=logits_out, labels=y_output)`

You are **calling softmax twice**. From TensorFlow [documentation](https://www.tensorflow.org/api_docs/python/tf/nn/sparse_softmax_cross_entropy_with_logits):

> WARNING: This op expects unscaled logits, since it performs a softmax on logits internally for efficiency. Do not call this op with the output of softmax, as it will produce incorrect results.

Have a nice day :)

**Comment by nfmcclure** [comment]
Hi @davidenunes ,  Thanks for noticing this.  Fixed two places I could find them.  Please reopen if you notice any more.
